### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by miku0805** [criyle/go-judge#57] [Q&A] [answered]
invalid semicolon separator in query,大佬这个报错怎么解决啊= =

**Comment by criyle** [comment, marked as answer]
也许是请求路径多了一个 `,` ，可以检查一下

**Reply by miku0805** [reply]
大佬，就是我用postman向/file发送了一个post请求，用是body里面的form-data，然后服务器里报错
ERRORzap@v0.1.0/zap.go:87invalid semicolon separator in query{"status": 400, "method": "POST", "path": "/file", "query": "con
tent={
    "cmd": [{
        "args": ["/usr/bin/g++", "a.cc", "-o", "a"],
        "env
": ["PATH=/usr/bin:/bin"],
        "files": [{
            "content": ""
        }
, {
            "name": "stdout",
            "max": 10240
        }, {
 %
20          "name": "stderr",
            "max": 10240
        }],
      %
20 "cpuLimit": 10000000000,
        "memoryLimit": 104857600,
        "procLimit": 50,
       %2
0"copyIn": {
            "a.cc": {
                "content": "#include <iostream%
3E%%nusing namespace std;%%nint main() {%%nint a, b;%%ncin >> a >> b;%%ncout << a + b << endl;%%n}"
     %
20      }
        },
        "copyOut": ["stdout", "stderr"],
        "copyOutCached
": ["a.cc", "a"],
        "copyOutDir": "1"
    }]
}", "ip": "14.21.28.196", "user-agent": "PostmanRuntime/7.31.1"
, "latency": "65.353µs"}
github.com/gin-contrib/zap.GinzapWithConfig.func1
/home/runner/go/pkg/mod/github.com/gin-contrib/zap@v0.1.0/zap.go:87
github.com/gin-gonic/gin.(*Context).Next
/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.8.2/context.go:173
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.8.2/gin.go:616
github.com/gin-gonic/gin.(*Engine).ServeHTTP
/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.8.2/gin.go:572
net/http.serverHandler.ServeHTTP
/opt/hostedtoolcache/go/1.20.1/x64/src/net/http/server.go:2936
net/http.(*conn).serve
/opt/hostedtoolcache/go/1.20.1/x64/src/net/http/server.go:1995
2023/03/15 12:40:30 http: URL query contains semicolon, which is no longer a supported separator; parts of the query may be stripped when parsed; see golang.org/issue/2
5192
然后我发送get请求到/file是正常的，其他/config也是能正常访问,我想知道是我哪里弄错了么

**Reply by criyle** [reply]
日志有 query 是以 query string 传入的参数，并不是 form data

**Reply by miku0805** [reply]
解决了，谢谢佬
